src/app.js

```diff
@@ -1 +1,117 @@
+[]    De naam van het land
+[]    De vlag van dat land
+[]    De zin: Has a population of [amount] people
+[]    De landen zijn gesorteert op populatie, van laag naar hoog;
+[]    De land-namen moeten worden weergegeven in een kleur die overeenkomt met het continent waar het land in ligt. Tip: maak hier een aparte functie voor die een regio-naam verwacht en bepaalt welke kleur het land moet krijgen. Een land ligt meestal in één van de volgende vijf contintenten, maar uitzonderingen kunnen voorkomen:
+
+[]    Africa: blauw
+[]    Americas: groen
+[]    Asia: rood
+[]    Europe: geel
+[]    Oceania: paars
+
+    Stappenplan
+    Let op: het is uitdagender om jouw eigen stappenplan te maken. Als je niet zo goed weet waar je moet beginnen, kun je onderstaand stappenplan gebruiken:
+
+[v]  1. Installeer en importeer Axios;
+            * https://www.npmjs.com/package/axios
+[v]  2. Neem de documentatie van de REST Countries API goed door. Welk endpoint heb je nodig om informatie over alle landen op te halen?
+            * https://restcountries.com/v2/all
+[v]  3. Schrijf een asynchrone functie die, met behulp van Axios, een GET-request maakt naar het juiste endpoint. Log de response in de console en bestudeer de data goed: hoe is het opgebouwd?
+[v]  4. Probeer eens om de naam van het allereerste land te loggen in de console, welk pad moet je hiervoor volgen?
+[v]  5. Maak een <ul>-tag in de HTML die je als referentie kunt gebruiken in jouw JavaScript bestand;
+[v]  6. Zorg ervoor dat de naam van het allereerste land wordt weergegeven als <li>-tag in het lijst-element op de pagina;
+            * Injecteren met de map-methode
+[v]  7. Zorg er nu ook voor dat de populatie (Has a population of [amount] people) daaronder wordt weergegeven;
+[v]  8. Schrijf een aparte functie die één regio-naam verwacht, en op basis van deze regio de correcte kleur-naam als string teruggeeft. Gebruik deze, om de naam van het land in de juiste kleur weer te geven op de pagina. Tip: zorg ervoor dat je CSS-classes maakt voor alle regio-kleuren!
+[v]  9. Breidt de <li>-tag uit met een <img>-tag om zo ook de url van de meegegeven vlag-afbeelding weer te kunnen geven;
+[v]  10.Gebruik de map-methode om over de array met landen heen te mappen, en zo een <li>-element te maken voor álle landen;
+[v]  11.Zorg er ten slotte voor dat je de response data eerst sorteert op populatie, van laag naar hoog, voor je eroverheen mapt om de landen op de pagina weer te geven.
+        Bonusopdrachten
+
+[v]    Lees paragraaf 8.2 op EdHub door en kijk hoe je dit kunt toepassen in deze opdracht;
+[v]    Make it look nice! 😍
+    */
+const listToHtml = document.getElementById('countyList');
+const errorMessage = document.getElementById('error');
+
+async function fetchCountries() {
+    try {
+        // het request maken
+        const result = await axios.get('https://restcountries.com/v2/all');
+        const countries = result.data.sort((a, b) => {
```

Comment:
Je kunt dit verkorten door het volgende te doen:
```suggestion
        const countries = result.data.sort(a, b) => a.population - b.population;
```